sched/irq/irq_csection.c

```diff
@@ -615,13 +613,14 @@ void restore_critical_section(void)
    * followed by context switching.
    */
 
-  FAR struct tcb_s *tcb = this_task();
+  FAR struct tcb_s *tcb;
   int me = this_cpu();
 
   /* Adjust global IRQ controls.  If irqcount is greater than zero,
    * then this task/this CPU holds the IRQ lock
    */
 
+  tcb = current_task(me);
```

Comment:
@hujun260 
Please tell us why did you change this line in this PR.

Reply:
this_task() is a function call that involves disabling interrupts, and since restore_critical_section is always called in an interrupt-disabled context, there's no need to disable interrupts again.

Therefore, to save time and achieve the same effect, I directly call tcb = current_task(me) instead of tcb = this_task().

Reply:
@masayuki2009 do you have more concern

Reply:
@hujun260 
Please add a separate commit (not a PR) for it.

Reply:
> @hujun260 Please add a separate commit (not a PR) for it.

ok

Reply:
@hujun260 
Thanks for dividing the commit.
However, I think the new commit message is still unclear and has long lines.

So, could you change the 2nd commit message as follows?

```
Author: hujun5 <hujun5@xiaomi.com>
Date:   Tue Jun 11 08:05:26 2024 +0800

    sched: irq: Change this_task to current_task(me) in restore_critical_section()
    
    this_task() is a function call that involves disabling interrupts and this_cpu().
    Since restore_critical_section is always called in an interrupt-disabled context,
    there's no need to disable interrupts again. Therefore, to save time and achieve
    the same effect, I directly call tcb = current_task(me) instead of tcb = this_task().
    
    Signed-off-by: hujun5 <hujun5@xiaomi.com>
```

Reply:
ok